File: Tutorial4-4ComponentBuildersMultipleModules/src/main/java/com/example/tutorial4_4componentbuildersmultiplemodules/MyApplication.java
package com.example.tutorial4_4componentbuildersmultiplemodules;

import android.app.Application;

import com.example.tutorial4_4componentbuildersmultiplemodules.di.AppComponent;
import com.example.tutorial4_4componentbuildersmultiplemodules.di.AppComponentWithBind;
import com.example.tutorial4_4componentbuildersmultiplemodules.di.AppModule;
import com.example.tutorial4_4componentbuildersmultiplemodules.di.DaggerAppComponent;
import com.example.tutorial4_4componentbuildersmultiplemodules.di.DaggerAppComponentWithBind;
import com.example.tutorial4_4componentbuildersmultiplemodules.di.SensorControllerModule;
import com.example.tutorial4_4componentbuildersmultiplemodules.di.ToastMakerModule;

public class MyApplication extends Application {

    private AppComponent appComponent;

    private AppComponentWithBind appComponentWithBuilder;


    @Override
    public void onCreate() {
        super.onCreate();

        appComponent = DaggerAppComponent
                .builder()
                .appModule(new AppModule(this))
                .sensorControllerModule(new SensorControllerModule())
                .toastMakerModule(new ToastMakerModule())
                .build();


        appComponentWithBuilder = DaggerAppComponentWithBind
                .builder()
                .application(this)
                .build();

    }


    public AppComponent getAppComponent() {
        return appComponent;
    }

    public AppComponentWithBind getAppComponentWithBuilder() {
        return appComponentWithBuilder;
    }
}

File: Tutorial4-4ComponentBuildersMultipleModules/src/main/java/com/example/tutorial4_4componentbuildersmultiplemodules/MyApplication.kt
package com.example.tutorial4_4componentbuildersmultiplemodules

import android.app.Application
import com.example.tutorial4_4componentbuildersmultiplemodules.di.*

class MyApplication : Application() {

    lateinit var appComponent: AppComponent
        private set

    lateinit var appComponentWithBuilder: AppComponentWithBind
        private set

    override fun onCreate() {
        super.onCreate()

        appComponent = DaggerAppComponent
                .builder()
                .appModule(AppModule(this))
                .sensorControllerModule(SensorControllerModule())
                .toastMakerModule(ToastMakerModule())
                .build()
        appComponentWithBuilder = DaggerAppComponentWithBind
                .builder()
                .application(this)
                .build()
    }

}